util/src/RawMemoryPool.cpp

```diff
@@ -41,7 +41,9 @@ void RawMemoryPool::allocatePool(int32_t initialChunksNum, int32_t maxChunksNum)
   maxChunksNum_ = maxChunksNum;
   pool_ = make_shared<boost::lockfree::queue<char*, boost::lockfree::fixed_sized<true>>>(maxChunksNum_);
   for (int32_t i = 0; i < initialChunksNum_; ++i) {
-    pool_->push(new char[chunkSize_]);
+    char* chunk = new char[chunkSize_];
```

Comment:
should we check if new returned Null here (in case)? We would be pushing empty memory in pool in that case.

Reply:
Operator new throws std::bad_alloc exception in case it fails allocating memory